File: Beecrowd/6-Paradigmas/1286-Motoboy-5/1286.cs
using System;

public class Solution{
    public static void Main(){

        int pedidos, m;

        while(true){
            pedidos = int.Parse(Console.ReadLine());
            if(0 ==pedidos)break;
            int[] tmp = new int[pedidos];
            int[] pizzas = new int[pedidos];
            int maxPizza = int.Parse(Console.ReadLine());
            for(int i =0;i<pedidos;i++){
                string[] aux = Console.ReadLine().Split(' ');
                tmp[i] = int.Parse(aux[0]);
                pizzas[i] = int.Parse(aux[1]);
            }

            int[,] dp = new int[pedidos+1, maxPizza+1];
            for(int i = 0; i < pedidos; i++){
                for(int j = 0; j <= maxPizza; j++){
                    if(j >= pizzas[i])  
                        dp[i+1, j] = Math.Max(dp[i, j], dp[i, j - pizzas[i]] + tmp[i]);
                    else
                        dp[i+1, j] = dp[i, j];  
                }
            }


            Console.WriteLine("{0} min.", dp[pedidos, maxPizza]);

        }
    }

}

File: Beecrowd/6-Paradigmas/1286-Motoboy-5/1286.py



while(True):
    pedidos=int(input())
    if(pedidos==0):break
    tmp = [0]*pedidos
    pizzas= [0]*pedidos
    maxPizza = int(input())
    for i in range(pedidos):
        tmp[i], pizzas[i] = map(int, input().split())

    dp = [[0] * (maxPizza + 1) for _ in range(pedidos + 1)]
    for i in range(pedidos):
        for j in range(maxPizza+1):
            if(j >= pizzas[i]) :
                dp[i+1][j] = max(dp[i][j], dp[i][j - pizzas[i]] + tmp[i])
            else:
                dp[i+1][j] = dp[i][j] 


    print(f"{dp[pedidos][maxPizza]} min.")

